GAE/src/org/waterforpeople/mapping/app/web/rest/QuestionRestService.java

```diff
@@ -448,4 +420,21 @@ private Question newQuestion(QuestionDto dto) {
         final Question result = questionDao.save(q);
         return result;
     }
+
+    private void attachAnyOptions(QuestionDto dto, List<QuestionOptionDto> qoResults) {
```

Comment:
Slightly less efficient, since we create & discard a List for every type except option, but ok.